tests/common.rs

```diff
@@ -125,8 +125,18 @@ pub async fn clean(storage: Storage<KeyTest>, path: impl AsRef<Path>) -> Result<
     fs::remove_dir_all(path).map_err(Into::into)
 }
 
-pub async fn close_storage(storage: Storage<KeyTest>) -> Result<()> {
-    std::thread::sleep(std::time::Duration::from_millis(100));
+const MAX_WAIT_CYCLES: usize = 10;
+const WAIT_DELAY: std::time::Duration = std::time::Duration::from_millis(20);
+
+pub async fn close_storage(storage: Storage<KeyTest>, expected_files: &[&PathBuf]) -> Result<()> {
```

Comment:
You can generalize this approach: create method that takes a predicate as an argument and wait in loop for it to return `true`. In this form, the method can be used in many places by passing various predicates.

Reply:
I've added separate fn for this (but it's still used only here, since other tests doesn't have obvious predicates to use)